### DISCUSSION TITLE
Floating Windows: dimension and position

### DISCUSSION THREAD

**Discussion by 24unix** [asmvik/yabai#774] [Ideas]
Would it be hard to implement ja feature in float mode to show the current window geometry like eg KDE can do?

yabai monitores if a window is moved or resized, isn't it?

**Comment by asmvik** [comment]
Not really sure what you mean by "show the current window geometry"? You can already retrieve the dimensions with the query system. Do you mean like a visual wireframe-like mode?

Yabai does know when a window is moved and resized, yes.

**Comment by 24unix** [comment]
I mean like so:
![Screenshot_20201226_120919](https://user-images.githubusercontent.com/36622631/103150360-b5181e80-4773-11eb-84ac-e3addab4ed2a.png)

**Comment by asmvik** [comment]
So the first thing that comes to mind is that we do not have the necessary information at hand. To replicate the functionality we would need to keep a cache of the current window frame so that we can diff it with the new frame. The notifications do not forward this information to us, so we would have to perform two requests through the AX API to get the information. At this point we need to create a window/context for drawing and display it.

Hard to tell what the performance would be like when multiple windows are moved/resized at the same time.

**Comment by 24unix** [comment]
Oh, I didn't mean that much.

Only if you have eg:
  "frame":{
                "x":27.0000,
                "y":87.0000,
                "w":3159.0000,
                "h":1637.0000
},

That those values would be shown, no diff to the former state.

So like
     27, 87
    (3159x1637)

**Comment by asmvik** [comment]
Should be doable. I'll probably experiment a bit to see what its like.

**Comment by asmvik** [comment]
This seems to be working fine if the floating window is moved or resized using the mouse, but will require a more sophisticated solution to work for windows that are moved through other means, as there is not a separate event for begin/end move/resize.

**Comment by 24unix** [comment]
That would be finde for me, as I thought only about float mode, I guess nobody is interested in the dimensions if many windows are moving at the same time.
